### PR TITLE
[ACA-3390]Add startProcessWithApplication method on StartProcessPage

### PR DESCRIPTION
**Pull request by cristinaj**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/testing/src/lib/process-services/pages/start-process.page.ts [hunk 1]
```diff
@@ -145,9 +152,21 @@ export class StartProcessPage {
         return new FormFields();
     }
 
+    async checkSelectProcessField(): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsNotVisible(this.disabledSelectProcessDropdown);
```
Comment (arditdomi) [inline lib/testing/src/lib/process-services/pages/start-process.page.ts:156]: can we have a name like checkSelectProcessDropdownIsEnabled?

File: lib/testing/src/lib/process-services/pages/start-process.page.ts [hunk 2]
```diff
@@ -34,6 +35,7 @@ export class StartProcessPage {
     processDefinitionOptionsPanel: ElementFinder = element(by.css('div[class*="mat-autocomplete-panel"]'));
 
     dropdownPage = new DropdownPage();
+    applicationDropdownPage = new DropdownPage(element(by.css('[data-automation-id*="start-process-app"] div[class="mat-select-arrow"]')));
```
Comment (arditdomi) [inline lib/testing/src/lib/process-services/pages/start-process.page.ts:38]: just a small improvement. Now that we have 2 dropdown we could rename to processDefinitionDropdown and applicationDropdownPage. There is no need to wrap the method like we do here https://github.com/Alfresco/alfresco-ng2-components/blob/74aadb162ecc2a83c364f83be4030db550b022a9/lib/testing/src/lib/process-services/pages/start-process.page.ts#L90 instead we could directly use `this.applicationDropdownPage.selectOption` and `this.processDefinitionDropdown.selectOption`

File: lib/testing/src/lib/process-services/pages/start-process.page.ts [hunk 3]
```diff
@@ -145,9 +152,21 @@ export class StartProcessPage {
         return new FormFields();
     }
 
+    async checkSelectProcessField(): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsNotVisible(this.disabledSelectProcessDropdown);
+    }
+
     async startProcess({name, processName }) {
```
Comment (eromano) [inline lib/testing/src/lib/process-services/pages/start-process.page.ts:159]: avoid use object in the firm of the method use two input parameters, we never use this kind of declaration

File: lib/testing/src/lib/process-services/pages/start-process.page.ts [hunk 4]
```diff
@@ -145,9 +152,21 @@ export class StartProcessPage {
...
         await this.enterProcessName(name);
         await this.selectFromProcessDropdown(processName);
         await this.clickStartProcessButton();
     }
+
+    async startProcessWithApplication({name, applicationName, processName }) {
```
Comment (eromano) [inline lib/testing/src/lib/process-services/pages/start-process.page.ts:165]: avoid use object in the firm of the method use three input parameters, we never use this kind of declaration

File: lib/testing/src/lib/process-services/pages/start-process.page.ts (hunk 3 above)
Comment (cristinaj) [reply lib/testing/src/lib/process-services/pages/start-process.page.ts:159]: This is how it was implemented previously. I've seen this method it's not much used, so i can refactor.

File: lib/testing/src/lib/process-services/pages/start-process.page.ts (hunk 1 above)
Comment (cristinaj) [reply lib/testing/src/lib/process-services/pages/start-process.page.ts:156]: Renamed

File: lib/testing/src/lib/process-services/pages/start-process.page.ts (hunk 2 above)
Comment (cristinaj) [reply lib/testing/src/lib/process-services/pages/start-process.page.ts]: I've renamed dropdownPage to processDefinitionDropdown and applicationDropdownPage to applicationDropdown. About the wrapping, it seems selectOption method is used in some tests too and there i wouldn't want to use the class member directly in the test.

**Pull request merged**